### PR TITLE
[dev branch] TS gazelle extension

### PR DESCRIPTION
**Pull request by alexeagle**
This will be a long-lived feature branch while we get to a working MVP

### CODE REVIEW

**Comment by sluongng** [comment]
Hi folks,

Just a drive by comment but I thought you guys might be interested in my proposal over here https://github.com/bazelbuild/bazel-gazelle/issues/1189.

Thanks for creating this 🤗

File: gazelle/tests/simple_import_generated/BUILD.out [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+genrule(
+    name = "generator",
+    outs = ["genlib.js"],
+    cmd = "echo 'export const ANSWER = 42;' > $@",
+)
+
+ts_project(
+    name = "simple_import_generated",
+    srcs = ["main.ts"],
+    deps = ["//:genlib.js"],
```
Comment (alexeagle) [inline gazelle/tests/simple_import_generated/BUILD.out:13]: in reality this isn't allowed: deps of ts_project must provide DeclarationInfo

**Review by alexeagle — COMMENTED** [review]
very confusing, but in TypeScript `rootDirs` is not the plural of `rootDir` - they have totally different meanings.
`rootDir` affects what files go in the program
`rootDirs` is for the module resolver to apply an extra pathmapping to treat two different directories as logical siblings.

It's a problem where we can develop here and have wrong typescript semantics, and nothing tells us. I think we need to extend the test harness so you also ask TypeScript whether the tsconfig/files in a given test fixture have the result you expect.

**Comment by alexeagle** [comment]
Now that we've decided to split TypeScript into its own ruleset, we should move this to aspect-build/rules_ts - I'll start on it.

**Comment by jbedard** [comment]
Replaced with https://github.com/aspect-build/rules_ts/pull/4

**Pull request closed without merging**